crates/solver/src/solver/http_solver/settlement.rs

```diff
@@ -217,17 +217,12 @@ fn match_prepared_and_settled_orders(
             let prepared = prepared_orders
                 .get(index)
                 .ok_or_else(|| anyhow!("invalid order {}", index))?;
-            match prepared.id {
-                crate::liquidity::LimitOrderUid::OrderUid(_) => {}
-                crate::liquidity::LimitOrderUid::ZeroEx(_) => {
-                    if let Some(internalizable) =
-                        settled.exec_plan.as_ref().map(|plan| plan.internal)
-                    {
-                        ensure!(
-                            !internalizable,
-                            "liquidity orders are not allowed to be internalizable"
-                        )
-                    }
+            if prepared.is_liquidity_order() {
+                if let Some(internalizable) = settled.exec_plan.as_ref().map(|plan| plan.internal) {
+                    ensure!(
+                        !internalizable,
+                        "liquidity orders are not allowed to be internalizable"
+                    )
```

Comment:
One bug fixed here (I know I could have fixed it by using field `is_liquidity_order` but I wanted to kill that field).